inttest/kuberouter/kuberouter_hairpin_test.go

```diff
@@ -77,10 +79,15 @@ func (s *KubeRouterHairpinSuite) TestK0sGetsUp() {
 			},
 		} {
 			t.Run(test.desc, func(t *testing.T) {
-				output, err := ssh.ExecWithOutput(s.Context(), fmt.Sprintf("%s --connect-timeout 5 -sS http://%s", curl, test.dnsName))
-				if !assert.NoError(t, err) || !assert.Contains(t, output, "Thank you for using nginx.") {
-					t.Log(output)
-				}
+				err = wait.PollImmediate(5*time.Second, 2*time.Minute, func() (bool, error) {
+					output, err := ssh.ExecWithOutput(s.Context(), fmt.Sprintf("%s --connect-timeout 5 -sS http://%s", curl, test.dnsName))
```

Comment:
unrelated to this diff in particular, but feels weird to have a constant string `curl` on the function level, let's make it either variable or move to the top level?